src/main/java/com/kobylynskyi/graphql/codegen/kotlin/KotlinGraphQLTypeMapper.java

```diff
@@ -0,0 +1,145 @@
+                Utils.isNotBlank(mappingContext.getSubscriptionReturnType())) {
+            // in case it is subscription and subscriptionReturnType is set
+            return getGenericsString(mappingContext, mappingContext.getSubscriptionReturnType(), computedTypeName);
+        }
+        // Consider only Java-kotlin OR Java-Scala cross language calls
+        if (Boolean.TRUE.equals(mappingContext.getUseOptionalForNullableReturnTypes()) && !namedDefinition.isMandatory()) {
+            if (!computedTypeName.startsWith(KOTLIN_UTIL_LIST) && !computedTypeName.startsWith(JAVA_UTIL_LIST)) {
+                // append `?` (except java list and kotlin list)
+                computedTypeName = getOptionString(mappingContext, computedTypeName);
+            }
+        }
+
+        if (computedTypeName.startsWith(KOTLIN_UTIL_LIST) &&
+                Utils.isNotBlank(mappingContext.getApiReturnListType())) {
+            // in case it is query/mutation, return type is list and apiReturnListType is set
+            return computedTypeName.replace(KOTLIN_UTIL_LIST, mappingContext.getApiReturnListType());
+        }
+        if (Utils.isNotBlank(mappingContext.getApiReturnType())) {
+            // in case it is query/mutation and apiReturnType is set
+            return getGenericsString(mappingContext, mappingContext.getApiReturnType(), computedTypeName);
+        }
+        return getTypeConsideringPrimitive(mappingContext, namedDefinition, computedTypeName);
+    }
+
+    @Override
+    public boolean isPrimitive(String scalaType) {
+        return isKotlinPrimitive(scalaType);
+    }
+
+    @Override
+    public String getGenericsString(MappingContext mappingContext, String genericType, String typeParameter) {
+        return String.format("%s<%s>", genericType, typeParameter);
+    }
+
+    @Override
+    public String mapDirectiveArgumentValue(MappingContext mappingContext, Argument dirArg, String argumentValueFormatter) {
+        return valueMapper.map(mappingContext, dirArg.getValue(), null, argumentValueFormatter);
+    }
+
+    @Override
+    public boolean addModelValidationAnnotationForType(String possiblyPrimitiveType) {
+        return false;
+    }
+
+    public static String defaultValueKotlinPrimitive(String kotlinType) {
+        switch (kotlinType) {
+            case "Long":
+                return "0L";
+            case "Float":
+                return "0F";
+            case "Double":
+                return "0D";
+            case "Char":
+                return "0.toChar()";
+            case "Boolean":
+                return "false";
+            case "Int":
+            case "Byte":
+            case "Short":
+            default:
+                return "0";
+        }
+    }
+
+    private String getOptionString(MappingContext mappingContext, String typeParameter) {
+        return typeParameter + KotlinGraphQLTypeMapper.KOTLIN_UTIL_OPTIONAL;
```

Comment:
We should avoid the "optional" terminology here. Kotlin has nullable types and non-nullable types.